### PR TITLE
feat: add float, date, datetime object spec

### PR DESCRIPTION
**Pull request by CJHwong**
Reason
--------------
Supporting more type checking

Changes
--------------
add spec and validators for float, date object, date/time object

Test Scope
--------------
Covered by test cases.

Checks
--------------
- [x] Unit tests are included or not applicable.

### CODE REVIEW

File: data_spec_validator/spec/defines.py [hunk 1]
```diff
@@ -9,13 +9,16 @@
...
+FLOAT = 'float'
 DIGIT_STR = 'digit_str'  # URL params cannot distinguish from strings and numbers
 STR = 'str'
 BOOL = 'bool'
 JSON = 'json'
 JSON_BOOL = 'json_bool'
 LIST = 'list'
 DICT = 'dict'
+DATE_OBJECT = 'date_obj'
+DATETIME_OBJECT = 'datetime_obj'
```
Comment (Falldog) [inline data_spec_validator/spec/defines.py:21]: suggest remove `_OBJECT` suffix

Comment (CJHwong) [reply data_spec_validator/spec/defines.py:21]: There already has a DATE spec for validating date strings.
Removing _OBJECT suffix would be a breaking change to this package.

To smooth the transition, I'd change the original spec name to `DATE_STR` while still providing `DATE` as the alias to it, and replace the DATE_OBJECT by DATE in the next major release.

Comment (Falldog) [reply data_spec_validator/spec/defines.py:21]: 原來如此

File: data_spec_validator/spec/validators.py [hunk 2]
```diff
@@ -259,6 +273,26 @@ def validate(value, extra, data) -> Tuple[bool, Union[Exception, str]]:
...
+        info = '' if ok else TypeError(f'{repr(value)} is not a date object')
+        return ok, info
+
+
+class DatetimeObjectValidator(BaseValidator):
+    name = DATETIME_OBJECT
+
+    @staticmethod
+    def validate(value, extra, data):
+        ok = type(value) is datetime.datetime
```
Comment (Falldog) [inline data_spec_validator/spec/validators.py:291]: replace it by isinstance?
滿多狀況會有 inherit datetime object 的做法的，但如果直接判斷 type 的話，就喪失相容的機會了
(note: str 也是)

Comment (CJHwong) [reply data_spec_validator/spec/validators.py:291]: 我覺得這樣不錯 👍 

但改用 isinstance 的話就喪失嚴格檢查 type 的能力了， @kilikkuo 對這點有疑慮嗎？
沒問題的話我會把全部用 is 做的檢查改成用 isinstance

Comment (Falldog) [reply data_spec_validator/spec/validators.py:291]: 我覺得未必要全改啦 可以針對比較常會需要繼承修改的處理即可

Comment (Falldog) [reply data_spec_validator/spec/validators.py:291]: 或是 有人開需求再處理就好 哈

Comment (kilikkuo) [reply data_spec_validator/spec/validators.py:291]: 我認為讓 dsv 的 builtin check 有明確的限制檢查的範圍就好.
其他衍伸的繼承格式, 需要的 project 可以寫 custom check

**Comment by kilikkuo** [comment]
幫忙修改一下 README 吧 :)

**Review by kilikkuo — APPROVED** [review]
Please add more information in README

**Pull request merged**